diff -up Python-2.7.3/Lib/test/test_gdb.py.gdb-autoload-safepath Python-2.7.3/Lib/test/test_gdb.py
--- Python-2.7.3/Lib/test/test_gdb.py.gdb-autoload-safepath	2012-04-30 12:55:43.372351709 -0400
+++ Python-2.7.3/Lib/test/test_gdb.py	2012-04-30 12:58:24.355339144 -0400
@@ -111,6 +111,12 @@ class DebuggerTests(unittest.TestCase):
                     # Disable this:
                     'set print entry-values no',
 
+                    # Recent GDBs will only auto-load scripts from certain safe
+                    # locations.
+                    # Turn off this protection during the build to ensure that
+                    # our -gdb.py script can be loaded (rhbz#817072):
+                    'set auto-load safe-path /',
+
                     'run']
         if cmds_after_breakpoint:
             commands += cmds_after_breakpoint
